src/commandline_options.cpp

```diff
@@ -343,9 +343,9 @@ commandline_options::commandline_options (const std::vector<std::string>& args)
 	if (vm.count("core"))
 		core_id = vm["core"].as<std::string>();
 	if (vm.count("config-dir"))
-		userdata_dir = vm["config-dir"].as<std::string>(); //TODO: complain and remove
+		throw boost::program_options::error(formatter() << "config-dir is deprecated, use userdata-dir instead.");
```

Comment:
The `formatter()` is not needed when there's just a single string. You can pass that string directly, `boost::program_options::error("foo")` (you may need to wrap it in std::string but I doubt that).

The error message should probably be translated.

The man page should be updated. doc/man/wesnoth.6

What's the most recent release where these options _weren't_ marked as deprecated? That is, how many release cycles have they been deprecated for?

Before you do this, though: I'm not sure we _need_ to remove these, at all. I think this case is a perfect fit for https://wiki.wesnoth.org/CompatibilityStandards#Deprecation_levels_-_When_to_remove_deprecated_features level 1 "Deprecated indefinitely". It adds no complexity to keep the option parsing and handling. So why are we removing these options at all? What would be the downside if we closed #1466 as won't fix?

Reply:
I'm actually not sure how to mark these strings for translation. The other command-line errors are not translated but `po::options_description`s are.

I'm not sure this is needed either but just trying to close #1466. @Vultraz Can we close #1466 without resorting to this?